src/java.base/share/classes/java/security/PEMEncoder.java

```diff
@@ -0,0 +1,348 @@
+/*
+ * Copyright (c) 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package java.security;
+
+import jdk.internal.javac.PreviewFeature;
+import sun.security.pkcs.PKCS8Key;
+import sun.security.util.DerOutputStream;
+import sun.security.util.DerValue;
+import sun.security.util.Pem;
+import sun.security.x509.AlgorithmId;
+
+import javax.crypto.*;
+import javax.crypto.spec.PBEKeySpec;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.security.cert.*;
+import java.security.spec.AlgorithmParameterSpec;
+import java.security.spec.PKCS8EncodedKeySpec;
+import java.security.spec.X509EncodedKeySpec;
+import java.util.Objects;
+import java.util.concurrent.locks.ReentrantLock;
+
+/**
+ * {@code PEMEncoder} is used for encoding Privacy-Enhanced Mail (PEM) data.
+ * PEM is a textual encoding used to store and transfer security
+ * objects, such as asymmetric keys, certificates, and certificate revocation
+ * lists (CRL).  It is defined in RFC 1421 and RFC 7468.  PEM consists of a
+ * Base64-formatted binary encoding enclosed by a type-identifying header
+ * and footer.
+ *
+ * <p> Encoding may be performed on Java Cryptographic Extension (JCE) objects
+ * that implement {@link DEREncodable}.
+ *
+ * <p> Encrypted private key PEM data can be built by encoding with a
+ * {@code PEMEncoder} instance returned by {@linkplain #withEncryption(char[])}
+ * or by encoding an {@link EncryptedPrivateKeyInfo} .
```

Comment:
Suggest rewording as: "Private keys can be encrypted and encoded by configuring a `PEMEncoder` with the `withEncryption` method, which takes a password and returns a new `PEMEncoder` instance configured to encrypt the key with that password. Alternatively, a private key encrypted as an `EncryptedKeyInfo` object can be encoded directly to PEM by passing it to the `encode` method."